### ISSUE TITLE
False Positive S1854/no-dead-store: increment and decrement operators

### ISSUE THREAD

**Issue by Kinrany** [SonarSource/SonarTS#757]
**I want to report a bug.**

**SonarTS version:** 1.8.0  
**Node.js version:** 6.14.1  
**TypeScript version:** 3.0.3  

**Rule key:** "no-dead-store"

**Reproducer** 
```typescript
function test() {
    const a = { x : 0 };
    const b = "x"; // False positive here
    a[b]++;
    return a;
}
```

**Logs**
```
ERROR: src/main.ts[3, 11]: Remove this useless assignment to local variable "b".
```

**Expected behavior**
No error since `b` is being used to index `a`.

---

Workaround: replace `a[b]++;` with `a[b] += 1;`
